File: backend/app/schemas/futuro.py
"""Schemas Pydantic para el módulo de Futuro (renovaciones y plan anual).

Cubre el Epic 9 de la spec: feed de licitaciones adjudicadas renovables
con fecha estimada de término de contrato, y el plan anual de compras
de los organismos.
"""

from datetime import datetime
import uuid

from pydantic import BaseModel


class RenovacionResponse(BaseModel):
    """Licitación adjudicada con potencial de renovación."""

    model_config = {"from_attributes": True}

    licitacion_codigo: str
    nombre: str
    organismo_nombre: str | None
    monto_estimado: float | None
    fecha_adjudicacion: datetime | None
    duracion_estimada_meses: int | None
    fecha_estimada_termino_contrato: datetime | None
    dias_para_termino: int | None


class RenovacionesListResponse(BaseModel):
    """Respuesta paginada del feed de renovaciones."""

    total: int
    page: int
    page_size: int
    items: list[RenovacionResponse]


class PlanAnualLineaResponse(BaseModel):
    """Línea del plan anual de compras de un organismo."""

    model_config = {"from_attributes": True}

    id: uuid.UUID
    ano: int
    codigo_organismo: int
    descripcion: str
    unspsc_codigo: str | None
    unspsc_nombre: str | None
    monto_estimado: float | None
    moneda: str
    mes_estimado: int | None
    modalidad: str | None
    status: str
    licitacion_codigo: str | None
    created_at: datetime


class PlanAnualListResponse(BaseModel):
    """Respuesta paginada del plan anual."""

    total: int
    page: int
    page_size: int
    items: list[PlanAnualLineaResponse]

File: frontend/src/types/futuro.ts
export interface RenovacionItem {
  licitacion_codigo: string
  nombre: string
  organismo_nombre: string | null
  monto_estimado: number | null
  fecha_adjudicacion: string | null
  duracion_estimada_meses: number | null
  fecha_estimada_termino_contrato: string | null
  dias_para_termino: number | null
}

export interface RenovacionesListResponse {
  total: number
  page: number
  page_size: number
  items: RenovacionItem[]
}

export interface PlanAnualLinea {
  id: string
  ano: number
  codigo_organismo: number
  descripcion: string
  unspsc_codigo: string | null
  unspsc_nombre: string | null
  monto_estimado: number | null
  moneda: string
  mes_estimado: number | null
  modalidad: string | null
  status: "planificada" | "publicada" | "adjudicada" | "cancelada"
  licitacion_codigo: string | null
  created_at: string
}

export interface PlanAnualListResponse {
  total: number
  page: number
  page_size: number
  items: PlanAnualLinea[]
}
